compose/compose-file/index.md

```diff
@@ -1186,8 +1195,41 @@ volumes:
 
 > **Note:** The long syntax is new in v3.2
 
-See [Docker Volumes](/engine/userguide/dockervolumes.md) and
-[Volume Plugins](/engine/extend/plugins_volume.md) for more information.
+
+#### Volumes for services, swarms, and stack files
+
+When working with services, swarms, and `docker-stack.yml` files, keep in mind
+that the tasks (containers) backing a service can be deployed on any node in a
+swarm, which may be a different node each time the service is updated.
+
+In the absence of having named volumes with specified sources, Docker creates an
+anonymous volume for each task backing a service. Anonymous volumes do not
+persist after the associated containers are removed.
+
+If you want your data to persist, use a named volume and a volume driver that
+is multi-host aware, so that the data is accessible from any node. Or, set
+constraints on the service so that its tasks are deployed on a node that has the
+volume present.
+
+As an example, the `docker-stack.yml` file for the
+[votingapp sample in Docker
+Labs](https://github.com/docker/labs/blob/master/beginner/chapters/votingapp.md) defines a service called `db` that runs a `postgres` database. It is
+configured as a named volume in order to persist the data on the swarm,
+_and_ is constrained to run only on `manager` nodes. Here is the relevant snip-it from that file:
```

Comment:
We should avoid using constraints to `manager` nodes in our examples. Generally, it's recommended to not put load on manager nodes (as that may result in your manager running out of resources). While it's perfectly valid to do so (if you know what you're doing), probably best to not use it in our examples, unless for a specific reason